back/app/services/participants_service.rb

```diff
@@ -106,63 +104,76 @@ def project_participants(project)
   def folder_participants_count(folder)
     Rails.cache.fetch("#{folder.cache_key}/participant_count", expires_in: 1.day) do
-      participant_ids = projects_participants(folder.projects).pluck(:id)
-      known_author_hashes = participant_ids.flat_map do |id|
-        folder.projects.ids { |project_id| Idea.create_author_hash(id, project_id, true) }
-      end
-      participant_ids.size + projects_anonymous_count(folder.projects, known_author_hashes) + projects_everyone_count(folder.projects)
+      Analytics::FactParticipation
+        .where(dimension_project_id: folder.projects.pluck(:id))
+        .select(:participant_id)
+        .distinct
+        .count
     end
   end
 
-  def projects_anonymous_count(projects, known_author_hashes)
-    anonymous_idea_hashes = Idea.where(project: projects, anonymous: true)
-      .where.not(author_hash: known_author_hashes)
-      .distinct.pluck(:author_hash)
-    anonymous_comment_hashes = Comment.joins(:idea)
-      .where(anonymous: true)
-      .where(idea: { project: projects })
-      .where.not(author_hash: known_author_hashes + anonymous_idea_hashes)
-      .distinct.pluck(:author_hash)
-    anonymous_idea_hashes.size + anonymous_comment_hashes.size
-  end
-
-  def projects_everyone_count(projects)
-    Idea.where(project: projects, author: nil, anonymous: false).size
-  end
-
   def projects_participants(projects, options = {})
     since = options[:since]
-    actions = options[:actions] || PARTICIPANT_ACTIONS
+    actions = options[:actions] || PROJECT_PARTICIPANT_ACTIONS
     ideas = Idea.where(project: projects)
     participants = ideas_participants(ideas, options)
-    # voting
+
+    # Voting
     if actions.include? :voting
       baskets = Basket.submitted
       baskets = baskets.where(phase: Phase.where(project: projects))
       baskets = baskets.where('created_at::date >= (?)::date', since) if since
       participants = participants.or(User.where(id: baskets.select(:user_id)))
     end
+
     # Polling
     if actions.include? :polling
       poll_responses = Polls::Response.where(phase: Phase.where(project: projects))
       poll_responses = poll_responses.where('created_at::date >= (?)::date', since) if since
       participants = participants.or(User.where(id: poll_responses.select(:user_id)))
     end
+
     # Volunteering
     if actions.include? :volunteering
       volunteering_users = User.joins(volunteers: [:cause])
       volunteering_users = volunteering_users.where(volunteering_causes: { phase: Phase.where(project: projects) })
       participants = participants.or(User.where(id: volunteering_users))
     end
+
+    # Event attending
+    if actions.include? :event_attending
+      event_attendances = Events::Attendance.where(event: Event.where(project: projects))
+      event_attendances = event_attendances.where('created_at::date >= (?)::date', since) if since
+      participants = participants.or(User.where(id: event_attendances.select(:attendee_id)))
```

Comment:
Just a few suggestions, you can throw them away:
```suggestion
      event_attendances = Events::Attendance
        .joins(:event).where(events: { project: projects })
        .where(created_at: since..)
      event_attendees = User.where(id: event_attendances.select(:attendee_id))
      participants = participants.or(event_attendees)
```
The `JOIN` does not change anything performance-wise compared to the nested query in this case, just a different style. I kinda like the range notation (`since..`). It degrades well if `since` is nil.